apps/onboarding/src/wallet/method-filter.ts

```diff
@@ -0,0 +1,53 @@
+import { extractMethodId, normalizeMethodId } from '@app/blockchain/utils'
+import { Err, normalizeAddress, Ok, Result } from '@celo/base'
+import { CeloContract } from '@celo/contractkit'
+import { BaseWrapper } from '@celo/contractkit/lib/wrappers/BaseWrapper'
+import { RawTransaction } from '@celo/contractkit/lib/wrappers/MetaTransactionWallet'
+import { Contract } from 'web3-eth-contract'
+
+export interface TransactionWithMetadata {
+  raw: RawTransaction,
+  methodId: string,
+  methodName: string,
+  contractName: string
+}
+
+export class MethodFilter {
+  private allowed: Map<string, boolean> = new Map()
```

Comment:
If `allowed` and `methodName` are always 1:1.. does it make sense to create a new class that wraps both?
```
class Method {
  string methodName
  boolean allowed
}
```
and then replace both with:
 `private methods: Map<string, Method> = new Map()`

Reply:
Yup good callout I actually simplified it further with the caveat of some data duplication by storing the contract name along side each method name, but simpler computationally and more readable.